RESETPASSWORD: Fix dubble GET request

eunjuhuss opened the pull request:
#### Description:
This PR is fix for the dubble triggering of GET request when reset password page is loading

#### Summary:
- Remove `csrf_token` from useEffect array

---
- before fix
<img width="654" alt="Screenshot 2021-06-01 at 13 37 25" src="https://user-images.githubusercontent.com/44289056/120435183-b4686180-c37d-11eb-92ed-a0b9cbe398a9.png">
<img width="654" alt="Screenshot 2021-06-01 at 13 37 35" src="https://user-images.githubusercontent.com/44289056/120435187-b5998e80-c37d-11eb-95f4-b39c7fcddb12.png">

- after fix
<img width="654" alt="Screenshot 2021-06-02 at 08 32 36" src="https://user-images.githubusercontent.com/44289056/120435063-8f73ee80-c37d-11eb-8569-34551c5b1554.png">
<img width="654" alt="Screenshot 2021-06-02 at 08 32 22" src="https://user-images.githubusercontent.com/44289056/120435057-8daa2b00-c37d-11eb-9be3-a1017acddf1b.png">


---


 

#### For reviewer:
- [ ] Read the above description
- [ ] Reviewed the code changes
- [ ] Navigate to the branch (pulled the latest version)
- [ ] Run the code and been able to execute the expected function
- [ ] Check any styling on both desktop and mobile sizes

njons commented on src/login/components/LoginApp/ResetPassword/ResetPasswordMain.js:
```diff
@@ -1,18 +1,17 @@
-  const csrf_token = useSelector(state => state.resetPassword.csrf_token);
 
   useEffect(() => {
     dispatch(getResetPassword());
-  }, [csrf_token]);
+  }, []);
```
Maybe there is an unexpected re-render of this component, because I can still see two GET_RESET_PASSWORD and GET_RESET_PASSWORD_SUCCESS.

in redux:
![Screenshot 2021-06-03 at 15 45 55](https://user-images.githubusercontent.com/30963614/120655461-fdeea480-c482-11eb-9479-5466dc89165a.png)

in network tab:
![Screenshot 2021-06-03 at 15 37 59](https://user-images.githubusercontent.com/30963614/120655723-3f7f4f80-c483-11eb-9211-1f0b981fe993.png)
![Screenshot 2021-06-03 at 15 38 18](https://user-images.githubusercontent.com/30963614/120655676-34c4ba80-c483-11eb-856d-7afe9db76eb6.png)

njons reviewed (commented):
I am still seeing two api calls to /reset-password.

eunjuhuss commented:
- Removed ResetPasswordMain component to prevent re-rendering on loading of page emailLinkSent
- Added  dispatch(getResetPassword()) in ResetPasswordForm

njons reviewed (commented):
From what I can see when I run the branch, altering the components does not remove the extra API calls (although its a nice clean up of the jsx). 
Looking closer at the calls themselves, it is not clear to me what purpose they serve as all you get in response is a crsf_token. So I looked into removing them and as far as I can tell the flow works as intended.

To me, this still works:
- resetPasswordSaga.js: 
    - removing lines 15-36 
    - line 53 get crsf_token from config: csrf_token: state.config.csrf_token
- login-rootSaga.js:
    - removing lines 11 and 13

If the calls are needed the way you have implemented them then maybe a revisit of how the backend is intended to work is needed. 
In particular line 13 in `login-rootSaga.js`  is puzzling to me: 
`takeEvery(postResetPasswordActions.POST_RESET_PASSWORD_FAIL, getResetPasswordConfig)`

eunjuhuss commented:
Yes, you are right, we have Get login config so we don't need an extra GET request from reset password to get token. so now I removed all GRT requests from reset password.
